### PR TITLE
shell: modules: kernel: add human readable uptime

### LINKED ISSUE #62543: shell: modules: kernel: Human readable uptime
**Is your enhancement proposal related to a problem? Please describe.**
Kernel uptime is shown in ms. An uptime of 13445303 ms is not easily readable.

**Describe the solution you'd like**
Keep the original functionality and add an option to print the uptime in e.g. day, hours, minutes, seconds, and milliseconds.
Maybe do like [uptime](https://www.man7.org/linux/man-pages/man1/uptime.1.html) -p but with higher resolution.

### PR DESCRIPTION
**Pull request by jakub-uC**
Added options: -p and --pretty to the kernel uptime command.

Fixes: #62543

### CODE REVIEW

File: subsys/shell/modules/kernel_service.c [hunk 1]
```diff
@@ -44,13 +44,46 @@ static int cmd_kernel_version(const struct shell *sh,
 	return 0;
 }
 
+#define SECONDS_FACTOR	1000u
```
Comment (jeppenodgaard) [inline subsys/shell/modules/kernel_service.c:47]: Nit: maybe use `MSEC_PER_SEC`

**Review by jeppenodgaard — APPROVED** [review]
LGTM. Only one nit.

**Comment by jakub-uC** [comment]
@henrikbrixandersen  @nordic-krch  please take a look

File: subsys/shell/modules/kernel_service.c [hunk 2]
```diff
@@ -44,13 +44,45 @@ static int cmd_kernel_version(const struct shell *sh,
...
+#define DAYS_FACTOR    (HOURS_FACTOR * 24)
+
 static int cmd_kernel_uptime(const struct shell *sh,
 			     size_t argc, char **argv)
 {
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	shell_print(sh, "Uptime: %u ms", k_uptime_get_32());
+	uint32_t milliseconds = k_uptime_get_32();
```
Comment (nordic-krch) [inline subsys/shell/modules/kernel_service.c:57]: what about using `k_uptime_get_64` as 32 bit limits to 49 days.

Comment (jakub-uC) [reply subsys/shell/modules/kernel_service.c]: @nordic-krch  done

File: subsys/shell/modules/kernel_service.c [hunk 3]
```diff
@@ -44,13 +44,45 @@ static int cmd_kernel_version(const struct shell *sh,
 	return 0;
 }
 
-static int cmd_kernel_uptime(const struct shell *sh,
-			     size_t argc, char **argv)
+#define MINUTES_FACTOR (MSEC_PER_SEC * 60)
+#define HOURS_FACTOR   (MINUTES_FACTOR * 60)
+#define DAYS_FACTOR    (HOURS_FACTOR * 24)
```
Comment (henrikbrixandersen) [inline subsys/shell/modules/kernel_service.c:49]: Would it make sense to add new definitions for these in `include/zephyr/sys_clock.h`? E.g. `SEC_PER_MIN`, `MIN_PER_HOUR`, ...

Comment (jakub-uC) [reply subsys/shell/modules/kernel_service.c:49]: sure

**Review by dcpleung — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**